include/dynd/callable.hpp

```diff
@@ -155,9 +155,9 @@ namespace nd {
 
     std::intptr_t get_narg() const { return get_type()->get_npos(); }
 
-    const ndt::type &get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
+    const ndt::type get_arg_type(std::intptr_t i) const { return get_type()->get_pos_type(i); }
 
-    const array &get_arg_types() const { return get_type()->get_pos_types(); }
+    nd::array get_arg_types() const { return get_type()->get_pos_types(); }
```

Comment:
Okay.  Regarding the const reference to the internal std::vector:  I'm not sure yet what the ownership invariants are.  If everything is a reference to the internal vector of the tuple type, what prevents the tuple from disappearing while something still holds a reference (I'm is usually programming in C, this might be an elementary thing in C++).

Is it an option to use only the move semantics you mentioned below, i.e. not use references at all?